hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockManager.java

```diff
@@ -2751,6 +2751,13 @@ public boolean checkBlockReportLease(BlockReportContext context,
       return true;
     }
     DatanodeDescriptor node = datanodeManager.getDatanode(nodeID);
+    if (node == null) {
+      final UnregisteredNodeException e = new UnregisteredNodeException(nodeID, null);
+      NameNode.stateChangeLog.error("BLOCK* NameSystem.getDatanode: " + "Data node " + nodeID +
```

Comment:
> I am confused if we need log here which upper method also do that when meet UnregisteredNodeException. Although it is DEBU level now, we could improve that to WARN, right?

Thanks @Hexiaoqiao for your review. I will wait for @ayushtkn reply and decide whether to update again.

Reply:
@tomscut you can remove this exception from here, considering it is being logged in the upper method.
>Although it is DEBU level now, we could improve that to WARN, right?

I think debug is also fine, this doesn't denote any problematic stuff, this can happen in normal scenario as well and the datanode will register subsequently

Reply:
> @tomscut you can remove this exception from here, considering it is being logged in the upper method.
> 
> > Although it is DEBU level now, we could improve that to WARN, right?
> 
> I think debug is also fine, this doesn't denote any problematic stuff, this can happen in normal scenario as well and the datanode will register subsequently

Hi @ayushtkn, do you mean I just remove this log?

Reply:
Yeps,
But if @Hexiaoqiao wants to change the log in the upper method to log. I am good with that as well

Reply:
> Yeps, But if @Hexiaoqiao wants to change the log in the upper method to log. I am good with that as well

Thank you very much for your comments. I will update the code.